test/performance/odp_l2fwd_perf.c

```diff
@@ -0,0 +1,1271 @@
+		    odp_pktin_queue(pktio->handle, pktio->in_qs, num_input_qs) !=
+		    (int)num_input_qs) {
+			ODPH_ERR("Error querying packet I/O input queues (%s)\n", pktio->name);
+			return false;
+		}
+
+		if (odp_pktout_queue(pktio->handle, pktio->out_qs, num_output_qs) !=
+		    (int)num_output_qs) {
+			ODPH_ERR("Error querying packet I/O output queues (%s)\n", pktio->name);
+			return false;
+		}
+
+		if (config->mtu > 0U) {
+			if (capa.set_op.op.maxlen == 0U) {
+				config->mtu = 0U;
+				ODPH_ERR("MTU setting not supported (%s)\n", pktio->name);
+				return false;
+			}
+
+			if (config->mtu < capa.maxlen.min_input ||
+			    config->mtu > capa.maxlen.max_input ||
+			    config->mtu < capa.maxlen.min_output ||
+			    config->mtu > capa.maxlen.max_output) {
+				config->mtu = 0U;
+				ODPH_ERR("Invalid MTU requested: %u (input min: %u, input max: %u,"
+					 " output min: %u, output max: %u, %s)\n",
+					 config->mtu, capa.maxlen.min_input, capa.maxlen.max_input,
+					 capa.maxlen.min_output, capa.maxlen.max_output,
+					 pktio->name);
+				return false;
+			}
+
+			config->orig_in_mtu = odp_pktin_maxlen(pktio->handle);
+			config->orig_out_mtu = odp_pktout_maxlen(pktio->handle);
+
+			if (odp_pktio_maxlen_set(pktio->handle, config->mtu, config->mtu) < 0) {
+				config->mtu = 0U;
+				ODPH_ERR("Error setting promiscuous mode (%s)\n", pktio->name);
+				return false;
+			}
+		}
+
+		if (config->is_promisc) {
+			if (capa.set_op.op.promisc_mode == 0U) {
+				config->is_promisc = 0U;
+				ODPH_ERR("Promiscuous mode setting not supported (%s)\n",
+					 pktio->name);
+				return false;
+			}
+
+			ret = odp_pktio_promisc_mode(pktio->handle);
+
+			if (odp_pktio_promisc_mode_set(pktio->handle, true) < 0) {
+				config->is_promisc = 0U;
+				ODPH_ERR("Error setting promiscuous mode (%s)\n", pktio->name);
+				return false;
+			}
+
+			config->orig_is_promisc = ret < 0 ? 1U : (uint8_t)ret;
+		}
+
+		if (odp_pktio_mac_addr(pktio->handle, pktio->src_mac.addr, ODPH_ETHADDR_LEN) < 0) {
+			ODPH_ERR("Error querying MAC address (%s)\n", pktio->name);
+			return false;
+		}
+
+		if (odp_pktio_start(pktio->handle) < 0) {
+			ODPH_ERR("Error starting packet I/O (%s)\n", pktio->name);
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static void print_mac(odph_ethaddr_t mac)
+{
+	for (int i = 0; i < ODPH_ETHADDR_LEN; ++i)
+		printf("%02x%s", mac.addr[i], i < ODPH_ETHADDR_LEN - 1 ? ":" : "");
+
+	printf("\n");
+}
+
+static odp_bool_t print_summary(const prog_config_t *config)
+{
+	const pktio_t *pktio;
+
+	printf("\nprogram options:\n================\n\n"
+	       "  interfaces:\n\n");
+
+	for (uint32_t i = 0U; i < config->num_ifs; ++i) {
+		pktio = &config->pktios[i];
+		printf("    %s:\n\n"
+		       "      handle:        0x%" PRIx64 "\n"
+		       "      src MAC:       ", pktio->name, odp_pktio_to_u64(pktio->handle));
+		print_mac(pktio->src_mac);
+		printf("      dst MAC:       ");
+		print_mac(pktio->dst_mac);
+		printf("      dst name:      %s\n"
```

Comment:
Minor thing but I'd perhaps print the name first instead of the handle.

Reply:
It is, it's the subject line for listed packet I/O entries.